app/ante/eth.go

```diff
@@ -375,3 +375,59 @@ func (issd EthIncrementSenderSequenceDecorator) AnteHandle(ctx sdk.Context, tx s
 
 	return next(ctx, tx, simulate)
 }
+
+// VirtualFrontierContractDecorator denies any transaction that tries to transfer funds to the virtual frontier contract address.
+type VirtualFrontierContractDecorator struct {
+	evmKeeper EVMKeeper
+}
+
+// NewVirtualFrontierContractDecorator creates a new VirtualFrontierContractDecorator instance.
+func NewVirtualFrontierContractDecorator(evmKeeper EVMKeeper) VirtualFrontierContractDecorator {
+	return VirtualFrontierContractDecorator{
+		evmKeeper: evmKeeper,
+	}
+}
+
+// AnteHandle checks if the transaction tries to transfer funds to the virtual frontier contract address.
+func (ctd VirtualFrontierContractDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	for _, msg := range tx.GetMsgs() {
+		msgEthTx, ok := msg.(*evmtypes.MsgEthereumTx)
+		if !ok {
+			return ctx, errorsmod.Wrapf(errortypes.ErrUnknownRequest, "invalid message type %T, expected %T", msg, (*evmtypes.MsgEthereumTx)(nil))
+		}
+
+		ethTx := msgEthTx.AsTransaction()
```

Comment:
optimize this snippet 

       to := ethTx.To()
        if to == nil || !ctd.evmKeeper.IsVirtualFrontierContract(ctx, *to) {
            continue // Skip if no recipient or not a VF contract
        }

Reply:
Thanks for your contrubution. Could you please explain me how changing that would help **optimize** the code?

Reply:
instead of 2 if check, you can just do one operation and it will check both.

Reply:
you call `ethTx.To()` 2 times, why not assign it to a variable and then do the checking.

Reply:
Sorry I put a comment here but idk why it gone or not submitted.

Please check the comment again to understand why I decided to not to merge the two statements
```golang
ethTx := msgEthTx.AsTransaction()

if ethTx.To() == nil {
	// contract creation tx => none of business
	continue
}

to := *ethTx.To()

if !ctd.evmKeeper.IsVirtualFrontierContract(ctx, to) {
	// not a VF contract => none of business
	continue
}

// reject if message value is not zero
if ethTx.Value().Sign() != 0 {
	return ctx, errorsmod.Wrapf(
		evmtypes.ErrProhibitedAccessingVirtualFrontierContract,
		"virtual frontier contract does not accept funds",
	)
}
```